backend/contributions/serializers.py

```
@@ -484,6 +484,19 @@ def get_contribution(self, obj):
 
+    def validate_contribution_date(self, value):
+        """
+        The contribution date is client-provided and flows into date-ranged
+        leaderboards once accepted, so future dates are rejected server-side
+        (small allowance for client clock skew).
+        """
+        from django.utils import timezone
+        from datetime import timedelta
+
+        if value and value > timezone.now() + timedelta(minutes=5):
+            raise serializers.ValidationError('Contribution date cannot be in the future.')
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)